src/components/common/Extensions/UserHubButton/UserHubButton.tsx

```diff
@@ -68,6 +70,15 @@ const UserHubButton: FC<UserHubButtonProps> = ({
       },
     );
 
+  const handleButtonClick = () => {
+    trackEvent({
+      event: 'custom_event',
+      category: 'User',
+      action: 'Click',
+      label: 'Open UserHub',
+    });
+  };
+
```

Comment:
This is not ok. Too many hardcoded values. Everything in here should be `enum` or otherwise a type. 

The only one that I can maybe accept as being a plain string is the `label`'s value.

Generally I'm not ok with patterns like this all strung about the codebase. Unless a super specific case calls for it, they should all be added to top-level / core components (buttons, forms, etc), not on the specific common components, as this way it's not maintainable.

Reply:
This has now been updated.